src/cpp/client.cpp

```diff
@@ -939,17 +961,15 @@ void Client::_unpack_dataset_metadata(DataSet& dataset,
 TensorBase* Client::_get_tensorbase_obj(const std::string& name)
 {
     std::string g_key = this->_build_tensor_key(name, true);
-
-    CommandReply reply = this->_redis_server->get_tensor(g_key);
+    GetTensorCommand cmd;
```

Comment:
I think if we put the parser functions as static member functions, we won't need to create this command.